lib/Bio/EnsEMBL/DataCheck/Checks/HGNCTypes.pm

```diff
@@ -0,0 +1,64 @@
+=head1 LICENSE
+
+Copyright [2018-2020] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::HGNCTypes;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'HGNCTypes',
+  DESCRIPTION    => 'Check that HGNC_curated_genes xrefs are on genes, _transcript are on transcript etc',
+  GROUPS         => ['core', 'xref'],
+  DATACHECK_TYPE => 'critical',
+  TABLES         => ['xref'],
+  PER_DB         => 1
+};
+
+sub tests {
+  my ($self) = @_;
+  #note these are looking for the *wrong* assignments
+  my %check_type =(
```

Comment:
I just checked in the production db, and none of these are current (http://production-services.ensembl.org/admin/ensembl_production_db/masterexternaldb/?q=HGNC&is_current=all).
Need to test "HGNC" => "Transcript" and "HGNC_trans_name" => "Gene" (assuming you keep the same logic - I think it'd be clearer if the hash had the expectation of what was correct, and the SQL condition did '<>' rather than '='. That would also test if anything is incorrectly assigned to Translations, which is not currently covered by this test.)